### ISSUE TITLE
Harness data source for application cannot retrieve app_id after app provisioned

### ISSUE THREAD

**Issue by arimaverick** [harness/terraform-provider-harness#136]
Hi,

We have a terraform module for harness components (comprises of workflow, pipeline, etc) for which we need to retrieve the app_id from data source inside the module. In order to test the module, we are provisioning the harness application and passing the name of the harness application in the next component module call (the component module can only accept application name for our generic usecase). However, we noticed that the data source cannot instantly retrieve the app_id from Harness after the resource application has been provisioned in the same terraform deployment.

### Terraform Version
1.1.8

### Affected Resource(s)
- data.harness_application

### Terraform Configuration Files
```hcl
resource "harness_application" "myapp" {
  name        = "test_harness_app"
  description = "This is my first Harness application"
}

data "harness_application" "myapp" {
    name = "test_harness_app"
    depends_on = [harness_application.myapp]
}

output "appi_id_from_ds" {
   value = data.harness_application.myapp.id == null ? "null app id" = data.harness_application.myapp.id
}

output "appi_id_from_resource_attr" {
   value = harness_application.myapp.id == null ? "null app id" = harness_application.myapp.id
}

```
output appi_id_from_resource_attr can produce the app_id but not the other output from data source appi_id_from_ds. 

### Expected Behavior
Data source should be able to retrieve the harness app_id in the same deployment where it provisions the harness app. 

### Actual Behavior
Null app_id is retrieved  from the data source

### Steps to Reproduce
Apply the above code

### Important Factoids
harness version 0.2.7

**Comment by arimaverick** [comment]
@micahlmartin I totally agree with you that we can use the resource attribute , however we have separated the application resource provisioning with that of its component provisioning. They are in two different modules for our specific use-case. we can use only data source inside the module which accepts the name of the application.

**Comment by micahlmartin** [comment]
But these are all still part of the same terraform execution?
